### PR TITLE
[cli] Cli can output cmac 

### PR DESCRIPTION
**Pull request by qinkunbao**
PSI will be a separate PR.

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes # (issue)

## Type of change (select or add applied and delete the others)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update
- [ ] Additional test coverage
- [ ] Code cleanup or just sync with upstream third-party crates

## Checklist

- [ ] Fork the repo and create your branch from `master`.
- [ ] If you've added code that should be tested, add tests.
- [ ] If you've changed APIs, update the documentation.
- [ ] Ensure the tests pass (see CI results).
- [ ] Make sure your code lints/format.

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -71,27 +73,30 @@ struct Opt {
     command: Command,
 }
 
-fn decrypt(opt: EncryptDecryptOpt) -> Result<()> {
+fn decrypt(opt: EncryptDecryptOpt, cmac: &mut CMac) -> Result<()> {
```
Comment (mssun) [inline cli/src/main.rs:76]: How about this? `fn decrypt(opt: EncryptDecryptOpt) -> Result<CMac>`

File: crypto/src/lib.rs [hunk 2]
```diff
@@ -75,15 +77,21 @@ impl AesGcm256Key {
         Self::default()
     }
 
-    pub fn decrypt(&self, in_out: &mut Vec<u8>) -> Result<()> {
+    pub fn decrypt(&self, in_out: &mut Vec<u8>) -> Result<[u8; CMAC_LENGTH]> {
```
Comment (mssun) [inline crypto/src/lib.rs:80]: You can use CMac instead of `[u8; CMAC_LENGTH]` .

File: cli/src/main.rs (hunk 1 above)
Comment (qinkunbao) [reply cli/src/main.rs]: Thanks. Fixed!

File: cli/src/main.rs [hunk 3]
```diff
@@ -69,71 +71,86 @@ enum Command {
 struct Opt {
     #[structopt(subcommand)]
     command: Command,
+
+    /// Whether to print cmac
+    #[structopt(short, long)]
+    cmac_flag: bool,
```
Comment (mssun) [inline cli/src/main.rs:77]: This flag should belong to the encrypt/decrypt sub-commands. Please define it in the `EncryptDecryptOpt` struct.

**Pull request closed without merging**